testdata/sqllogictests/functions/hashing.slt

```diff
@@ -0,0 +1,108 @@
+statement error
+select siphash(1, 2, 3);
+
+statement error
+select siphash(1, 2);
+
+statement ok
+select siphash(1);
+
+statement ok
+select siphash('000');
+
+statement ok
+select siphash(9001);
+
+statement ok
+select siphash(true);
+
+
+statement error
+select fnv(1, 2, 3);
+
+statement error
+select fnv(1, 2);
+
+statement ok
+select fnv(1);
+
+statement ok
+select fnv('000');
+
+statement ok
+select fnv(9001);
+
+statement ok
+select fnv(true);
+
+query I
+select siphash();
+----
+13715208377448023093
+
+query I
+select siphash(42);
+----
+8315904219845249920
+
+query I
+select siphash(3000);
+----
+14490819164275330428
+
+query I
+select siphash('42');
+----
+8771948186893062792
+
+query I
+select siphash([0, 100, 100, 300, 500, 800]);
+----
+3773389192103504674
+
+query I
+select fnv();
```

Comment:
I think this violate the type signature of `TypeSignature::Any(1)`. We either shouldn't allow this, or have the signature reflect this.

Reply:
shouldn't the function dispatcher prevent this? Will convert it into an error

Reply:
Unless we are bypassing something, I'd expect datafusion's planner to handle this. Maybe we should create a MRE and open up an issue with them.

Reply:
I haven't intentionally bypassed anything: you did more work in the function handling stuff recently than me.

Reply:
I agree that providing nothing and having it not error is a bit weird. I don't know if this is datafusion allowing that through, or us not propagating the signature upwards correctly. But I think this is still good to merge as is since I think fixing what's allowed based on the signature to be a separate scope.

Reply:
yeah, agreed.